ci/terraform/modules/txma-audit-queue/policy.tf

```diff
@@ -8,6 +8,18 @@ data "aws_iam_policy_document" "txma_audit_queue_access_policy_document" {
     actions   = ["sqs:SendMessage", ]
     resources = [aws_sqs_queue.txma_audit_queue.arn]
   }
+
+  statement {
+    effect = "Allow"
+    sid    = "AllowAccessToKeyForEncryptingPayloads"
+    actions = [
+      "kms:GenerateDataKey",
```

Comment:
Do we definitely need this permission? For SPOT the "kms:Descrypt" permission  was sufficient.

Reply:
I was following the guidance from https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-key-management.html (Configure KMS permissions for producers) and before I did this I got a "does not have kms:GenerateDataKey permissions" error in the logs.

Reply:
I'd definitely be interested in reduce permissions if possible, I wonder what we're doing in SPOT that's different.